File: hyeonsook95/baekjoon/9461.ts
var fs = require('fs');
var input = fs.readFileSync('dev/stdin').toString().split('\n');

var P = [1, 1, 1, 2, 2];
const T = parseInt(input.shift());

for (var t = 0; t < T; t++) {
    const n = input[t];

    if (n < P.length + 1) {
        console.log(P[n - 1]);
        continue;
    }

    while (n != P.length) {
        var idx = P.length - 1;
        P.push(P[idx] + P[idx - 4]);
    }
    console.log(P[P.length - 1]);
}

File: hyeonsook95/baekjoon/9461.py
P = [1, 1, 1, 2, 2]


def get_length(n):
    if n < len(P) + 1:
        return P[n - 1]

    while len(P) != n:
        idx = len(P) - 1
        P.append(P[idx] + P[idx - 4])
    return P[-1]


T = int(input())
for _ in range(T):
    print(get_length(int(input())))
